subprojects/language-java/src/main/java/org/gradle/api/tasks/javadoc/internal/JavadocSpec.java

```diff
@@ -60,10 +61,11 @@ public File getOptionsFile() {
         return optionsFile;
     }
 
-    public void setExecutable(String executable) {
+    public void setExecutable(@Nullable String executable) {
```

Comment:
Are you sure only this is nullable? It seems like the other properties are nullable, too, though I am not sure.

Reply:
I followed all call-sites of all other non-primitive setters and check that none of them are supplied nullable arguments. The `executable` is the only one that has been expected to be nullable.

Reply:
Hmm... Are they all set at all times as well? Since if you don't set them, then they'll be null. Anyway, given that this is internal code I think it is also fine to go ahead and merge as is.

Reply:
As far as I can tell, there is exactly one call to the `JavadocSpec` constructor in the whole codebase that calls all the setters. At the same time there are usages of the `JavadocSpec` instances that call getters and pass them into methods expecting non-nulls.